testthiscm.m

```diff
@@ -37,7 +37,7 @@
 witness_avg = trace(thiscm*thiswit);
 
 % Now test witness: Real and positive semidefinite
-witness_ok = isreal(thiswit)&& isPositiveDefinite(round(thiswit,10)) && (witness_avg - 1 < 0);
+witness_ok = isreal(thiswit)&& all(eig(round(thiswit,10))>=0) && (witness_avg - 1 < 0);
```

Comment:
You don't need to check all eigenvalues to be non-negative - just the least one should do! Although sorting will probably still use the same amount of resources as checking all of them.. (and is not a bottleneck what so ever!)